daprdocs/content/en/reference/components-reference/supported-secret-stores/kubernetes-secret-store.md

```diff
@@ -7,9 +7,27 @@ aliases:
   - "/operations/components/setup-secret-store/supported-secret-stores/kubernetes-secret-store/"
 ---
 
-## Summary
+## Create the Kubernetes Secret Store component
 
-Kubernetes has a built-in secrets store which Dapr components can use to retrieve secrets from. No special configuration is needed to setup the Kubernetes secrets store, and you are able to retrieve secrets from the `http://localhost:3500/v1.0/secrets/kubernetes/[my-secret]` URL. See this guide on [referencing secrets]({{< ref component-secrets.md >}}) to retrieve and use the secret with Dapr components.
+To setup a Kubernetes secret store create a component of type `secretstores.kubernetes`. See [this guide]({{< ref "setup-secret-store.md#apply-the-configuration" >}}) on how to create and apply a secretstore configuration. See this guide on [referencing secrets]({{< ref component-secrets.md >}}) to retrieve and use the secret with Dapr components.
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: mycustomsecretstore
+  namespace: default
+spec:
+  type: secretstores.kubernetes
+  version: v1
+  metadata:
+  - name: ""
+```
+{{% alert title="Warning" color="warning" %}}
+When Dapr is deployed to Kubernetes a secret store with name `kubernetes` is automatically provisioned. We discourage use of this secret store.
```

Comment:
I'd fix this wording. "Discourage use" makes people question why we even have it in the first place. I'd update this warning to say why not to use that one, and also when you would. This way people understand what each is for

Reply:
```suggestion
When Dapr is deployed to Kubernetes a secret store with name `kubernetes` is automatically provisioned. This is meant to streamline the use of the native secret store Kubernetes offers. However in most cases you would most likely want to define a Kubernetes secret store using the above YAML with a custom name. This will allow you to refer to the store in the application code by the custom name with no dependency on the component - allowing you to switch components when the application is deployed in different environments and potentially using different components for different secret store implementations.
```